lib/internal/freelist.js

```diff
@@ -9,18 +10,29 @@ class FreeList {
     this.name = name;
     this.ctor = ctor;
     this.max = max;
-    this.list = [];
+    this.list = ObjectSetPrototypeOf([], null);
   }
 
   alloc() {
-    return this.list.length > 0 ?
-      this.list.pop() :
-      ReflectApply(this.ctor, this, arguments);
+    const { list } = this;
+    const { length } = list;
+
+    if (length > 0) {
+      const lastIndex = length - 1;
+      const result = this.list[lastIndex];
+      delete this.list[lastIndex];
```

Comment:
I think `delete` is quite costy, maybe this performs better:
```suggestion
      this.list.length = lastIndex;
```

Reply:
Actually, it’s because `delete` of the last array element doesn’t update the `length` property, which is why [%Array.prototype.pop%](https://tc39.es/ecma262/#sec-array.prototype.pop) sets `this.length` after performing [DeletePropertyOrThrow](https://tc39.es/ecma262/#sec-deletepropertyorthrow).